### ISSUE TITLE
[BUG] Filling with unsigned ints in IntCategory axis raises TypeError

### ISSUE THREAD

**Issue by tcuisset** [scikit-hep/boost-histogram#898]
**Describe the bug**
 Filling an histogram with unsigned ints in IntCategory axis raises TypeError : "Only integer arrays supported when targeting integer axes". I believe this used to work in version 1.3

**Steps to reproduce**
~~~python
import boost_histogram as bh
import numpy as np
h = bh.Histogram(bh.axis.IntCategory([0, 1]))
h.fill(np.array([0], dtype="int32")) # works
h.fill(np.array([0], dtype="uint32")) # raises TypeError :  "Only integer arrays supported when targeting integer axes"
~~~

**Comment by henryiii** [comment]
It worked incorrectly for floats. I’d agree that uint would be nice to supported, though it will be doing a conversion internally since the fill routines are compiled with 64 bit signed ints. At the very least the error message can be improved.

**Comment by henryiii** [comment]
Fixed in #917.
